pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -169,11 +171,13 @@ func Setup(mgr ctrl.Manager, _ core.Args, _ logging.Logger) error {
 	if err != nil {
 		return fmt.Errorf("create discovery dm fail %w", err)
 	}
+	l := ctrl.Log.WithName("Application")
 	reconciler := Reconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("Application"),
-		Scheme: mgr.GetScheme(),
-		dm:     dm,
+		Client:     mgr.GetClient(),
+		Log:        l,
+		Scheme:     mgr.GetScheme(),
+		dm:         dm,
+		applicator: apply.NewAPIApplicator(mgr.GetClient(), logging.NewNopLogger().WithValues("Application")),
```

Comment:
The `logging.NewNopLogger()` will not print any thing, it's only used in tests.

BTW, we're refactoring our log lib to `klog/v2`, you can use it in the applicator

Reply:
Got it. I will move to klog/v2.

Reply:
It seems you haven't fixed

Reply:
I get the fix in another pr #1158.